# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Mohsi777999/mohsin-ali-flyrank/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [16]:
### My Lane

# I chose the Refresh / Content Opportunity Scoring lane because it focuses on identifying
# webpages that should be reviewed or refreshed first. Large websites often contain thousands of pages,
# making manual review inefficient. This project aims to prioritize pages that may need attention so
# content teams can focus their efforts where they are likely to have the greatest impact.
# I selected this lane because it combines data analysis with practical decision-making and
# aligns well with the starter notebooks and my goal of learning applied machine learning.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [15]:
# Decision, Action and Cost

# The decision I want to improve is identifying which webpages should be reviewed or refreshed first.
# The people who would use this output are content editors or SEO teams.
# They can use the ranked recommendations to decide which pages to update before spending time on lower-priority pages.
# A wrong recommendation could cause valuable pages to be ignored or waste time reviewing pages that do not actually need improvement.
# The goal is to support better decisions using data rather than manual guessing.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [14]:
import os, sys, subprocess

IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/Mohsi777999/mohsin-ali-flyrank"
REPO_DIR = "mohsin-ali-flyrank"

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)
    os.chdir(REPO_DIR)
    subprocess.run([sys.executable, "-m", "pip", "install", "-q", "-r", "requirements.txt"], check=True)
elif os.path.basename(os.getcwd()) == "notebooks":
    os.chdir("..")

import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")
print(df.shape)

# Basic statistics for the research question

print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

print("\nTrend Direction:")
print(df["trend_direction"].value_counts())

print("\nContent Types:")
print(df["content_type"].value_counts())

print("\nAverage CTR:")
print(round(df["ctr"].mean(), 2))

(30000, 44)
Rows: 30000
Columns: 44

Trend Direction:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Content Types:
content_type
keyword article       27207
feedly article         2096
comparison article      697
Name: count, dtype: int64

Average CTR:
0.51


The starter dataset contains **30,000 pages** with **44 features**, providing enough data for meaningful analysis.

I observed that **16,262 pages** have a **downward trend**, making declining content the largest group in the dataset. This suggests that identifying pages needing attention is an important problem.

I also found that **27,207 pages** are **keyword articles**, meaning this content type represents most of the dataset and is likely where many refresh decisions will be made.

Finally, the average CTR is approximately **0.51%**, showing that many pages receive relatively low click-through rates. These observations suggest that prioritizing pages for review could support better content management decisions.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [17]:
# This project can identify observed patterns and provide decision-support recommendations based on the available data.
# Any relationships found will be described as measured or directional observations.
# This project cannot prove why Google's rankings change or claim to predict Google's algorithm.
# The results should be used only to help prioritize content review decisions, not as causal proof.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.